CoreTableView/src/au/gov/asd/tac/constellation/views/tableview2/TableViewUtilities.java

```diff
@@ -1,354 +1,358 @@
+            }
+        });
+    }
+
+    /**
+     * Copy selection information from a table to a graph.
+     *
+     * @param table the table to retrieve data from.
+     * @param index the index of the selection.
+     * @param elementType the element type of the selection.
+     * @param graph the graph to copy to.
+     */
+    public static void copySelectionToGraph(final TableView<ObservableList<String>> table,
+            final Map<ObservableList<String>, Integer> index, final GraphElementType elementType, final Graph graph) {
+        PluginExecution.withPlugin(new SelectionToGraphPlugin(table, index, elementType)).executeLater(graph);
+    }
+
+    private static class ExportToCsvFilePlugin extends SimplePlugin {
+
+        private final File file;
+        private final TableView<ObservableList<String>> table;
+        private final boolean selectedOnly;
+
+        public ExportToCsvFilePlugin(final File file,
+                final TableView<ObservableList<String>> table,
+                final boolean selectedOnly) {
+            this.file = file;
+            this.table = table;
+            this.selectedOnly = selectedOnly;
+        }
+
+        @Override
+        public void execute(final PluginGraphs graphs, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+            try {
+                final String csvData = getTableData(table, true, selectedOnly);
+                try (final FileWriter fileWriter = new FileWriter(file)) {
+                    fileWriter.write(csvData);
+                }
+            } catch (IOException ex) {
+                throw new PluginException(PluginNotificationLevel.ERROR, ex);
+            }
+        }
+
+        @Override
+        public String getName() {
+            return EXPORT_TO_DELIMITED_FILE_PLUGIN;
+        }
+    }
+
+    private static class ExportToExcelFilePlugin extends SimplePlugin {
+
+        private final File file;
+        private final TableView<ObservableList<String>> table;
+        private final boolean selectedOnly;
+        private final String sheetName;
+
+        public ExportToExcelFilePlugin(final File file,
+                final TableView<ObservableList<String>> table,
+                final boolean selectedOnly, final String sheetName) {
+            this.file = file;
+            this.table = table;
+            this.selectedOnly = selectedOnly;
+            this.sheetName = sheetName;
+        }
+
+        @Override
+        public void execute(final PluginGraphs graphs, final PluginInteraction interaction, final PluginParameters parameters) throws InterruptedException, PluginException {
+            try (final SXSSFWorkbook workbook = new SXSSFWorkbook(SXSSFWorkbook.DEFAULT_WINDOW_SIZE)) {
+                File poiDir = new File(System.getProperty("java.io.tmpdir"), "poifiles");

```

Comment:
@serpens24 can we get a comment put here just to remind us years from now why we did this? Actually...I'm also wondering given you found this bug - would you like to commit this fix in the Apache poi library?

Reply:
@arcturus2 ... I've just spent the past hour trying to reproduce the original problem again, and haven't been able to (without my fix). I'm now at the point where I think the reason I was seeing the problem was how I was selecting the target directory sometimes ... in that I must have not double clicked to folder. .. because the default directory is the Network Shortcut folder that I've mentioned. I've discovered that clicking the desktop folder once is not selecting the folder, just highlighting it ... I suspect that is what I was doing. 
I'm going to suggest we discard my changes . ... as they are probably overkill brought about by me not double clicking.

Reply:
@serpens24 though I didn't mention it at the time I did feel this was overkill - happy to discard too - thanks mate!